code/modules/admin/spawn_panel/assets.dm

```diff
@@ -0,0 +1,56 @@
+/datum/asset/json/spawnpanel
+	name = "spawnpanel"
+
+/datum/asset/json/spawnpanel/generate()
+	var/list/data = list()
+	var/list/objects = typesof(/obj)
+	var/list/turfs = typesof(/turf)
+	var/list/mobs = typesof(/mob)
+
+	var/static/list/mapping_objects = typecacheof(list(
+		/obj/effect/mapping_helpers,
+  	/obj/effect/landmark,
+		/obj/effect/spawner,
+		/obj/effect/mob_spawn,
+		/obj/effect/holodeck_effect,
+		/obj/docking_port,
+		/obj/modular_map_connector,
+		/obj/modular_map_root,
+		/obj/pathfind_guy,
+	))
+
+	data["Objects"] = list()
+	data["Turfs"] = list()
+	data["Mobs"] = list()
+
+	for(var/item in objects)
+		var/obj/temp = item;
+		data["Objects"][item] = list(
+			"icon" = temp?.icon || "none",
+			"icon_state" = temp?.icon_state || "none",
```

Comment:
confused how this works tbh, shouldn't this be `temp::icon` and such?

also, `temp::icon_preview || temp::icon || "none"` and `temp::icon_state_preview || temp::icon_state || "none"` might be more appropriate to use, as `icon(_state)_preview` is intended for use for previews in menus and such. (note it's an `/obj` typepath, so it wouldn't be in turfs or mobs)